### PR TITLE
RD-1317 Apply group labels to deployments

### PR DESCRIPTION
**Pull request by tehasdf**
Copy group labels to the deployments, when the label is added.
Also remove it from the deployments, when the label is removed
from the group.

### CODE REVIEW

File: rest-service/manager_rest/rest/resources_v3_1/deployments.py [hunk 1]
```diff
@@ -575,21 +575,23 @@ def put(self, group_id):
...
-            group = sm.get(models.DeploymentGroup, group_id)
-        except manager_exceptions.NotFoundError:
-            group = models.DeploymentGroup(
-                id=group_id,
-                description=request_dict.get('description'),
-                created_at=datetime.now()
-            )
         with sm.transaction():
+            try:
+                group = sm.get(models.DeploymentGroup, group_id)
```
Comment (geokala) [inline rest-service/manager_rest/rest/resources_v3_1/deployments.py:580]: Do we deliberately not update the description when performing a put on a deployment group?

Comment (tehasdf) [reply rest-service/manager_rest/rest/resources_v3_1/deployments.py:580]: we do, in `_set_group_attributes`. You're right, no point doing it twice.

File: rest-service/manager_rest/storage/storage_manager.py [hunk 2]
```diff
@@ -99,7 +99,8 @@ def transaction(self):
 
         self._in_transaction = True
         try:
-            yield
+            with db.session.no_autoflush:
```
Comment (geokala) [inline rest-service/manager_rest/storage/storage_manager.py:102]: Out of interest, why this change?

File: rest-service/manager_rest/test/endpoints/test_deployment_groups.py [hunk 3]
```diff
@@ -437,6 +437,86 @@ def test_group_labels_for_deployments(self):
...
+    def test_add_labels_already_exist(self):
+        self.client.deployments.update_labels('dep1', [{'label2': 'value2'}])
+        self.client.deployment_groups.put(
+            'group1',
+            deployment_ids=['dep1'],
+        )
+        self.client.deployment_groups.put(  # doesn't throw
+            'group1',
+            labels=[{'label2': 'value2'}],
+        )
```
Comment (geokala) [inline rest-service/manager_rest/test/endpoints/test_deployment_groups.py:486]: Is it worth asserting on the what labels the deployment has here- a cheap operation but should alert us if we do something like duplicate labels with the same value?

**Review by geokala — DISMISSED** [review]

File: rest-service/manager_rest/test/endpoints/test_deployment_groups.py (hunk 3 above)
Comment (tehasdf) [reply rest-service/manager_rest/test/endpoints/test_deployment_groups.py:487]: there's an unique index on them, but why not!

**Review by geokala — DISMISSED** [review]

**Review by geokala — APPROVED** [review]
